tools/webpack-plugin/src/BacktracePluginV4.ts

```diff
@@ -20,62 +22,113 @@ export class BacktracePluginV4 implements WebpackPluginInstance {
     }
 
     public apply(compiler: Compiler) {
-        const assetDebugIds = new Map<string, string>();
+        const assetStats = new Map<string, AssetStats>();
 
         compiler.hooks.emit.tap(BacktracePluginV4.name, (compilation) => {
+            const logger = compilation.getLogger(BacktracePluginV4.name);
             for (const key in compilation.assets) {
                 let source = compilation.assets[key];
 
-                let debugId;
-                if (key.match(/.(c|m)?jsx?$/)) {
-                    debugId = crypto.randomUUID();
-                    assetDebugIds.set(key, debugId);
+                if (key.match(/\.(c|m)?jsx?$/)) {
+                    const debugId = crypto.randomUUID();
+                    const stats: AssetStats = { debugId };
+                    assetStats.set(key, stats);
+                    logger.log(`[${key}] generated debug ID ${debugId}`);
 
-                    source = this._sourceGenerator.addDebugIdToSource(source as never, debugId) as typeof source;
-                    source = this._sourceGenerator.addDebugIdCommentToSource(source as never, debugId) as typeof source;
+                    logger.time(`[${key}] inject source snippet`);
+                    try {
+                        source = this._sourceGenerator.addDebugIdToSource(source as never, debugId) as typeof source;
+                        logger.timeEnd(`[${key}] inject source snippet`);
+                        stats.sourceSnippet = true;
+                    } catch (err) {
+                        stats.sourceSnippet = err instanceof Error ? err : new Error('Unknown error.');
+                    }
+
+                    logger.time(`[${key}] inject sourcemap key`);
+                    try {
+                        source = this._sourceGenerator.addDebugIdCommentToSource(
+                            source as never,
+                            debugId,
+                        ) as typeof source;
+                        logger.timeEnd(`[${key}] inject sourcemap key`);
+                        stats.sourceComment = true;
+                    } catch (err) {
+                        stats.sourceComment = err instanceof Error ? err : new Error('Unknown error.');
```

Comment:
the time command from line 47 will never end. Should we add an error log about that? RIght now the user needs to check the source map comment however no one will do that in practice